metric/system/cgroup/util.go

```diff
@@ -276,6 +277,14 @@ the container as /sys/fs/cgroup/unified and start the system module with the hos
 				controllerPath = r.rootfsMountpoint.ResolveHostFS(filepath.Join("/sys/fs/cgroup/unified", path))
 			}
 
+			if tmp, ok := r.v2ControllerPathCache.Load(controllerPath); ok {
```

Comment:
Not sure I can follow the question :thinking: 
cgroup v2 can change. And regardless of this PR or the existing approach, data will be missed or incorrect. It will be missed or incorrect, as walking `controllerPath` to get all the `stat` files and reading the content of these files is happening independently. There is no change in this behaviour.
To address the concern that entries could stay forever in `r.v2ControllerPathCache` we could introduce a TTL. Would this be ok for you? And if so, which TTL would you consider? 5 minutes, 30 minutes, 1 hours, 2 hours or even 1 day?

Reply:
@kruskall with https://github.com/elastic/elastic-agent-system-metrics/pull/103/commits/0cb8e6e0ab759ce558b522a029900057be7c7aac I have added a timestamp to each cache entry to make sure entries do not stay forever in the cache. Does this address your concern?

Reply:
> My concern was about the system reusing PIDs causing the files in the cache to be inaccurate.

The porposed change does not have an effect on reused PIDs. At the moment of scanning all PIDs, eg. via `FetchPids()`, every PID might have some relation to a cgroup. And for cgroup v2 it is `controllerPath` that describes the dedicated cgroup v2. Usually multiple PIDs share the same cgroup v2. This proposed change reduces CPU usage load  by introducing a cache, to not look up every time `controllerPath` but cache its results (for 5 minutes).